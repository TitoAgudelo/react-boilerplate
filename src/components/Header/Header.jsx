import React from "react"
import styled from "styled-components"
import { motion } from "framer-motion";

const Header = () => {
  
  return (
    <HeaderContainer>
      <motion.div
        className="container"
        initial={{ scale: 0 }}
        animate={{ rotate: 360, scale: 1 }}
        transition={{
          type: "spring",
          stiffness: 260,
          damping: 40
      }}>
        <HeaderTitle testId="title">Posts</HeaderTitle>
      </motion.div>
      <HeaderText testId="author">By Tito Agudelo</HeaderText>
    </HeaderContainer>
  )
}

const HeaderContainer = styled.div`
  align-items: center;
  display: flex;
  height: 100px;
  justify-content: space-between;
  width: 100%;
`

const HeaderTitle = styled.p`
  color: ${({ theme: { colors } }) => colors.white };
  font-family: 'Macho SemiBold';
  font-size: 42px;
  line-height: 89.69%;
  margin-right: 10px;
`

const HeaderText = styled.p`
  color: ${({ theme: { colors } }) => colors.white };
  font-family: 'Macho Regular';
  font-weight: 500;
  font-size: 16px;
  line-height: 89.69%;
  text-transform: capitalize;
`

export default Header;
