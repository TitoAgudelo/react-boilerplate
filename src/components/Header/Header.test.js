import React from 'react';
import { shallow } from 'enzyme';
import Header from './Header';

let wrapped = shallow(<Header />);
let title = 'Posts';
let authorCopy = 'By Tito Agudelo';

describe('Header', () => {
  it('should render the Title Component correctly', () => {   
    expect(wrapped).toMatchSnapshot();
  });

  it('renders the Titles children', () => { 
    expect(wrapped.find('[testId="title"]').text()).toEqual(title);
  });

  it('renders the author children', () => { 
    expect(wrapped.find('[testId="author"]').text()).toEqual(authorCopy);
  });
});
