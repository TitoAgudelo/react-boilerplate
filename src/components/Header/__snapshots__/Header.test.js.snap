// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the Title Component correctly 1`] = `
<styled.div>
  <ForwardRef(MotionComponent)
    animate={
      Object {
        "rotate": 360,
        "scale": 1,
      }
    }
    className="container"
    initial={
      Object {
        "scale": 0,
      }
    }
    transition={
      Object {
        "damping": 40,
        "stiffness": 260,
        "type": "spring",
      }
    }
  >
    <styled.p
      testId="title"
    >
      Posts
    </styled.p>
  </ForwardRef(MotionComponent)>
  <styled.p
    testId="author"
  >
    By Tito Agudelo
  </styled.p>
</styled.div>
`;
